### ISSUE TITLE
ArrayIndexOutOfBoundsException when setting text on TextBox

### ISSUE THREAD

**Issue by vasso123** [mabe02/lanterna#390]
Example:
```java
public static void main(String[] args) throws IOException {
        terminalFactory = new DefaultTerminalFactory();
        terminal = terminalFactory.createTerminal();
        screen = new TerminalScreen(terminal);
        screen.startScreen();
        textGUI = new MultiWindowTextGUI(screen);
        window = new BasicWindow();

        TextBox tb = new TextBox();

        window.setComponent(tb);

        new Thread(() -> {
            try {
                Thread.sleep(3000);
            } catch (InterruptedException ex) {
                Logger.getLogger(TestTextBoxIssue.class.getName()).log(Level.SEVERE, null, ex);
            }
            textGUI.getGUIThread().invokeLater(() -> {
                tb.setText("\n");
            });
        }).start();

        textGUI.addWindowAndWait(window);
        screen.stopScreen();
    }

}
```
Only an issue when you set string containing only "\n" character.
Exception:
`Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: -1
	at java.util.ArrayList.elementData(ArrayList.java:418)
	at java.util.ArrayList.get(ArrayList.java:431)
	at com.googlecode.lanterna.gui2.TextBox.setText(TextBox.java:188)`

**Comment by avl42** [comment]
Good catch!

The problem here is because Java's  String.split( pattern )  does discard
all empty strings from end of its result.  That's how it can also return an
empty array.
Either we change behavior to not-eliminating empty lines at end, or we add
a workaround to fall back to a new String[1] {""}, if split returns a
String[0].

An even better approach would be to re-think the design of setText() and
addLine() methods of TextBox.   It seems to me as if addLine() already did
a better job handling newlines only if the TextBox is set for multiline
mode.  But then, addLine() also needs an update, because it goes recursive
on each "\n". that would break for longer texts.

If you need a quick workaround, exact (broken) behaviour of "secretly
multiline behavior despite singleline style), then change setText("\n") to
setText("") and then addLine("") .

I hope that mabe02 will give his opinion, on whether to fix the currently
broken SINGLE_LINE style or just catch the special case of split returning
a String[0].

PS: your testcase can be simplified a bit: it's not necessary to delay the
setText("\n") call. Just call it right after creation of TextBox().

**Comment by vasso123** [comment]
Thanks for the comment! :+1: 
As for the testcase, haven't bothered to make it perfect  :)

**Comment by mabe02** [comment]
Let's put in a quick fix for 3.0 and see if we can rethink this for 3.1

**Comment by mabe02** [comment]
That should do it
